homeassistant/config_entries.py

```diff
@@ -2726,6 +2726,39 @@ def is_matching(self, other_flow: Self) -> bool:
         """Return True if other_flow is matching this flow."""
         raise NotImplementedError
 
+    @property
+    def _reauth_entry_id(self) -> str:
+        """Return reauth entry id."""
+        if self.source != SOURCE_REAUTH:
+            raise ValueError(f"Source is {self.source}, expected {SOURCE_REAUTH}")
+        return self.context["entry_id"]  # type: ignore[no-any-return]
+
+    @callback
+    def _get_reauth_entry(self) -> ConfigEntry:
+        """Return the reauth config entry linked to the current context."""
+        if self.source != SOURCE_REAUTH:
+            raise ValueError(f"Source is {self.source}, expected {SOURCE_REAUTH}")
+        return cast(
```

Comment:
This cast is not safe, the config entry may have been deleted after the reauth flow started.
I'd suggest to check the return value of `config_entries.async_get_entry` and raise if it's None.

Reply:
For reference, the config entry is removed before the reauth flow is aborted: https://github.com/emontnemery/home-assistant/blob/9285f1eacde169f1960a24dfd92dcfcec1cba80f/homeassistant/config_entries.py#L1895-L1904

And, we don't even abort reconfigure flows (that's probably a bug).